src/Symfony/Component/Validator/ConstraintValidatorInterface.php

```diff
@@ -30,6 +33,10 @@ public function initialize(ExecutionContextInterface $context);
      *
      * @param mixed      $value      The value that should be validated
      * @param Constraint $constraint The constraint for the validation
+     *
+     * @throws UnexpectedTypeException       If the constraint or the value is incompatible (causes an unrecoverable error)
+     * @throws ConstraintDefinitionException If a constraint option is incompatible during runtime
+     * @throws ValidatorException            If any other validation related error occurred
```

Comment:
I am not convinced of this one either. Did you discover places where this exception is thrown?

Reply:
The validator exception is the generic one that all exceptions must extend from...

No I haven't searched.. The reason that this is added here is for custom constraints and exceptions..

```
class MyCustomValidationException extends ValidatorException{ }
```

```
class MyValidatorClass {
    public function validate() {
         ...
         throw new MyCustomValidationException();
         ...
    }
}
```

Reply:
IMO we should not document here what userland code *could* do (because then we would also have to add `\Exception` as that's something someone could also throw). But we should document here what we expect that could happen and what one should account for.

Reply:
forget for a second what symfony has implemented or not... this is an interface.. meaning an abstraction..

So this documentation should say what the implementations MUST throw in cases of exceptions.. it's not specific to what has been implemented or not.. In this way you are saying to whoever implements his own validator or even if someone submits it to symfony that the validator exceptions must always extend `ValidatorException`...

Reply:
Let's rephrase my previous comment then: custom validators must not throw any other exception than the `UnexpectedTypeException` if they want to comply with the interface.

Reply:
I can accept that as an opinion... Shall I open I proceed with the changes or wait for any other feedback?

Reply:
please proceed :)

Reply:
@gmponos Can you make the changes so that I can merge? Thank you.